api/pom.xml

```diff
@@ -65,4 +65,53 @@ under the License.
         </dependency>
     </dependencies>
 
+    <profiles>
+        <profile>
+            <id>mcp-enabled</id>
```

Comment:
It is better to name the profile as `java17`, as it is activated automatically when the JDK version is 17+.

Same for the other pom